libs/parallelism/execution/include/hpx/execution/algorithms/let_error.hpp

```diff
@@ -0,0 +1,282 @@
+#include <utility>
+#include <variant>
+
+namespace hpx { namespace execution { namespace experimental {
+    namespace detail {
+        template <typename PS, typename F>
+        struct let_error_sender
+        {
+            typename std::decay_t<PS> ps;
+            typename std::decay_t<F> f;
+
+            // Type of the potential values returned from the predecessor sender
+            template <template <typename...> class Tuple,
+                template <typename...> class Variant>
+            using predecessor_value_types =
+                typename hpx::execution::experimental::sender_traits<
+                    std::decay_t<PS>>::template value_types<Tuple, Variant>;
+
+            // Type of the potential errors returned from the predecessor sender
+            template <template <typename...> class Variant>
+            using predecessor_error_types =
+                typename hpx::execution::experimental::sender_traits<
+                    std::decay_t<PS>>::template error_types<Variant>;
+            static_assert(
+                !std::is_same<predecessor_error_types<hpx::util::pack>,
+                    hpx::util::pack<>>::value,
+                "let_error used with a predecessor that has an empty "
+                "error_types. Is let_error misplaced?");
+
+            template <typename E>
+            struct successor_sender_types_helper
+            {
+                using type = hpx::util::invoke_result_t<F,
+                    std::add_lvalue_reference_t<E>>;
+                static_assert(hpx::execution::experimental::is_sender<
+                                  std::decay_t<type>>::value,
+                    "let_error expects the invocable sender factory to return "
+                    "a sender");
+            };
+
+            // Type of the potential senders returned from the sender factor F
+            template <template <typename...> class Variant>
+            using successor_sender_types = hpx::util::detail::unique_t<
+                hpx::util::detail::transform_t<predecessor_error_types<Variant>,
+                    successor_sender_types_helper>>;
+
+            // The workaround for clang is due to a parsing bug in clang < 11
+            // in CUDA mode (where >>> also has a different meaning in kernel
+            // launches).
+            template <template <typename...> class Tuple,
+                template <typename...> class Variant>
+            using value_types = hpx::util::detail::unique_concat_t<
+                predecessor_value_types<Tuple, Variant>,
+                hpx::util::detail::concat_pack_of_packs_t<hpx::util::detail::
+                        transform_t<successor_sender_types<Variant>,
+                            value_types<Tuple, Variant>::template apply
+#if defined(HPX_CLANG_VERSION) && HPX_CLANG_VERSION < 110000
+                            >
+                    //
+                    >>;
+#else
+                            >>>;
+#endif
```

Comment:
This is strange! Does clang complain about `>>>`?

Reply:
Exactly. It's fixed in 11 though.